### PR TITLE
Ihp forum badges

### PR DESCRIPTION
**Pull request by Montmorency**
Pull request for [ihp-forum issue #1](https://github.com/digitallyinduced/ihp-forum/issues/1#issue-679555118). This PR just defines a Badges ENUM type and initializes a UserBadges table there is an Admin View of a UserBadges Table with the selectField. No UI side changes (icons etc to appear next to user profiles). Just Thought I would give it a go if this feature is still of interest ;).

### CODE REVIEW

File: Web/Controller/Sessions.hs [hunk 1]
```diff
@@ -9,4 +9,5 @@ instance Controller SessionsController where
     action CreateSessionAction = Sessions.createSessionAction @User
     action DeleteSessionAction = Sessions.deleteSessionAction @User
 
-instance Sessions.SessionsControllerConfig User where
\ No newline at end of file
+instance Sessions.SessionsControllerConfig User where
+    afterLoginRedirectPath = "/About"
```
Comment (mpscholten) [inline Web/Controller/Sessions.hs:13]: Is that action missing? :)

File: Application/Schema.sql [hunk 2]
```diff
@@ -30,7 +30,14 @@ CREATE TABLE topics (
...
+CREATE TABLE user_badges (
+    id UUID DEFAULT uuid_generate_v4() PRIMARY KEY NOT NULL,
+    user_id UUID NOT NULL,
+    badge badges NOT NULL
+);
 ALTER TABLE threads ADD CONSTRAINT threads_ref_topic_id FOREIGN KEY (topic_id) REFERENCES topics (id) ON DELETE NO ACTION;
+ALTER TABLE threads ADD CONSTRAINT threads_ref_user_id FOREIGN KEY (user_id) REFERENCES users (id) ON DELETE NO ACTION;
+ALTER TABLE comments ADD CONSTRAINT comments_ref_thread_id FOREIGN KEY (thread_id) REFERENCES threads (id) ON DELETE NO ACTION;
+ALTER TABLE comments ADD CONSTRAINT comments_ref_user_id FOREIGN KEY (user_id) REFERENCES users (id) ON DELETE NO ACTION;
+ALTER TABLE user_badges ADD CONSTRAINT user_badges_ref_user_id FOREIGN KEY (user_id) REFERENCES users (id) ON DELETE NO ACTION;
```
Comment (mpscholten) [inline Application/Schema.sql:43]: We should set the `ON DELETE` behavior to cascade here (badges should be deleted when the related user is deleted) 👍

**Comment by mpscholten** [comment]
Maybe we can also add the the badges to the user visible views already? :) We could use [bootstrap badges](https://getbootstrap.com/docs/4.5/components/badge/) for styling. What do you think?

File: Web/Controller/Sessions.hs (hunk 1 above)
Comment (Montmorency) [reply Web/Controller/Sessions.hs:13]: Woops this may have been from some earlier hacking round with ihp-forum. Can just drop it.

**Comment by Montmorency** [comment]
I like the badges styling. Also would a unique constraint on (user_id, badge) be a good check? You can only get one of each badge?

**Comment by mpscholten** [comment]
I think there could be people having e.g. `IhpStickerOwner` and `IhpContributor`, so basically we should allow people to have multiple badges :)

**Comment by Montmorency** [comment]
Oh definitely they should get more than one badge! I just meant that the Table shouldn't support record creation for (user_id, IhpStickerOwner) and then another (user_id, IhpStickerOwner).

**Comment by mpscholten** [comment]
Ok got it :) That makes sense 👍

**Comment by mpscholten** [comment]
Before merging we also need to add some kind of admin login in front of the admin backend :) Otherwise people could self-assign their badges :D

**Comment by Montmorency** [comment]
Haha yes can't be having that. I wasn't sure if you had a private forum branch with admin already set up. I will put in an Admin table and add ensureIsAdmin permissions.

**Comment by Montmorency** [comment]
Admin added. Admin table and Admin Authentication is active for both /admin/UserBadges and /admin/Admins controllers.

**Review by mpscholten — APPROVED** [review]
Thanks, looks good :)

**Pull request merged**